rules/coding-style/src/Rector/String_/UseClassKeywordForClassNameResolutionRector.php

```diff
@@ -50,23 +50,40 @@ public function getNodeTypes(): array
      */
     public function refactor(Node $node): ?Node
     {
-        if (substr_count($node->value, '::') !== 1) {
-            return null;
-        }
-        // a possible static call reference
         // @see https://regex101.com/r/Vv41Qr/1/
-        [$before, $possibleClass, $after] = Strings::split($node->value, '#([\\\\a-zA-Z0-9_\\x80-\\xff]*)::#');
+        $matches = Strings::matchAll($node->value, '#([\\\\a-zA-Z0-9_\\x80-\\xff]*)::#', PREG_PATTERN_ORDER);
+
+        $classNames = array_filter($matches[1], function (string $className) {
+            return class_exists($className);
+        });
 
-        if (! $possibleClass || ! class_exists($possibleClass)) {
-            return null;
+        if (empty($classNames)) {
```

Comment:
`empty` is very vague, as it includes many values null, false, 0...

Better use strict comparison
```suggestion
        if ($classNames === []) {
```

Reply:
ok,ok, I was kinda surprised that phpstan or rector didn't shout on me :D

Reply:
You're right, they should. I'll add a new rule :wink: 


I already added no isset/empty on objects: https://github.com/symplify/symplify/pull/1985